src/json_printer.sql

```diff
@@ -55,40 +70,65 @@ package body "JSON_PRINTER" as
       return str;
     end if;
   end llcheck;  
-
-  function escapeString(str varchar2) return varchar2 as
-    sb varchar2(32767) := '';
-    buf varchar2(40);
-    num number;
+  
+  -- escapes a single character. 
+  function escapeChar(ch char) return varchar2 deterministic is
```

Comment:
There would only be a single invocation of `x := y` on each invocation of this function: "After a matching WHEN clause is found, subsequent WHEN clauses are not executed." -- http://docs.oracle.com/cd/B19306_01/appdev.102/b14261/case_statement.htm